File: pdu/python/std_msgs/pdu_cdr_conv_UInt16.py
from .pdu_pytype_UInt16 import UInt16
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_UInt16(writer: CdrWriter, src: UInt16):
    writer.write_uint16(src.data)


def cdr_body_to_py_UInt16(reader: CdrReader, dst: UInt16):
    dst.data = reader.read_uint16()
    return dst


def py_to_cdr_UInt16(src: UInt16) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_UInt16(writer, src)
    return writer.bytes()


def cdr_to_py_UInt16(cdr_payload) -> UInt16:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = UInt16()
    return cdr_body_to_py_UInt16(reader, dst)

File: pdu/javascript/std_msgs/pdu_cdr_conv_UInt16.js
import { UInt16 } from './pdu_jstype_UInt16.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduUInt16Converter {
    /**
     * @param {PduCdrWriter} writer
     * @param { UInt16 } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint16(src.data);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { UInt16 } dst
     * @returns { UInt16 }
     */
    static cdr_body_to_js(reader, dst) {
        dst.data = reader.read_uint16();
        return dst;
    }

    /**
     * @param { UInt16 } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { UInt16 }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new UInt16());
    }
}
